### PR TITLE
Follow Hlint suggestions.

### PR DESCRIPTION
**Pull request by philderbeast**

### CODE REVIEW

File: test/Data/JSON/QQSpec.hs [hunk 1]
```diff
@@ -12,39 +12,32 @@ spec :: Spec
...
 
       it "parses empty arrays that include whitespace (regression test)" $ do
-        let Right value = parsedJson "[   ]"
-        value `shouldBe` JsonArray []
+        parsedJson "[   ]" `shouldBe` Right (JsonArray [])
 
       it "parses empty objects that include newlines (regression test)" $ do
-        let Right value = parsedJson "[\n]"
-        value `shouldBe` JsonArray []
+        parsedJson "[\n]" `shouldBe` Right (JsonArray [])
```
Comment (sol) [inline test/Data/JSON/QQSpec.hs:38]: `parsec` didn't provide `Eq ParseError` in the past (https://github.com/haskell/parsec/blob/master/ChangeLog.md#319). That's the reason why the old code used pattern matching here.

If you want to change this, then please also add `parsec >= 3.1.9` to the test suite.

File: test/Person.hs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{-# LANGUAGE OverloadedStrings, DeriveGeneric #-}
+{-# LANGUAGE DeriveGeneric #-}
```
Comment (sol) [inline test/Person.hs:1]: 👍

File: src/Data/JSON/QQ.hs [hunk 3]
```diff
@@ -137,4 +138,4 @@ unicodeChars = do
   d2 <- hexDigit
   d3 <- hexDigit
   d4 <- hexDigit
-  return $ u ++ [d1] ++ [d2] ++ [d3] ++ [d4]
+  return $ u ++ [d1] ++ [d2] ++ [d3] ++ [d4]
```
Comment (sol) [inline src/Data/JSON/QQ.hs:141]: 😨 👍

File: src/Data/JSON/QQ.hs [hunk 4]
```diff
@@ -115,19 +116,19 @@ symbol :: CharParser () String
...
-   <|> try (string "\\r" *> pure "\r")
-   <|> try (string "\\t" *> pure "\t")
-   <|> try (unicodeChars)
+       try (string "\\\"" $> "\"")
+   <|> try (string "\\\\" $> "\\")
+   <|> try (string "\\/" $> "/")
+   <|> try (string "\\b" $> "\b")
+   <|> try (string "\\f" $> "\f")
+   <|> try (string "\\n" $> "\n")
+   <|> try (string "\\r" $> "\r")
```
Comment (sol) [inline src/Data/JSON/QQ.hs:129]: 👍 but please also adjust the `base` lower bound accordingly

File: src/Data/Aeson/QQ.hs [hunk 5]
```diff
@@ -38,11 +38,11 @@ jsonExp txt =
 ---------
 toExp :: QQ.JsonValue -> ExpQ
 toExp (JsonString str) = [|String (T.pack str)|]
-toExp (JsonNull) = [|Null|]
+toExp JsonNull = [|Null|]
```
Comment (sol) [inline src/Data/Aeson/QQ.hs:41]: 👍

File: src/Data/Aeson/QQ.hs [hunk 6]
```diff
@@ -38,11 +38,11 @@ jsonExp txt =
...
 toExp (JsonObject objs) = [|object $jsList|]
     where
       jsList :: ExpQ
-      jsList = ListE <$> mapM objs2list (objs)
+      jsList = ListE <$> mapM objs2list objs
```
Comment (sol) [inline src/Data/Aeson/QQ.hs:45]: 👍

File: src/Data/JSON/QQ.hs [hunk 7]
```diff
@@ -39,22 +40,22 @@ jpValue = do
...
-jpBool = JsonBool <$> (string "true" *> pure True <|> string "false" *> pure False)
+jpBool = JsonBool <$> ((string "true" $> True) <|> (string "false" $> False))
 
 jpCode :: JsonParser
 jpCode = JsonCode <$> (string "#{" *> parseExp')
   where
     parseExp' = do
       str <- many1 (noneOf "}") <* char '}'
-      case (parseExp str) of
+      case parseExp str of
```
Comment (sol) [inline src/Data/JSON/QQ.hs:50]: 👍

File: src/Data/JSON/QQ.hs [hunk 8]
```diff
@@ -81,7 +82,7 @@ jpNumber = JsonNumber <$> do
 jpObject :: JsonParser
 jpObject = do
   list <- between (char '{') (char '}') (spaces *> commaSep jpHash)
-  return $ JsonObject $ list
+  return $ JsonObject list
```
Comment (sol) [inline src/Data/JSON/QQ.hs:85]: 👍

File: src/Data/JSON/QQ.hs [hunk 9]
```diff
@@ -115,19 +116,19 @@ symbol :: CharParser () String
 symbol = many1 (noneOf "\\ \":;><${}")
 
 commaSep :: CharParser () a -> CharParser () [a]
-commaSep p  = p `sepBy` (char ',')
+commaSep p  = p `sepBy` char ','
```
Comment (sol) [inline src/Data/JSON/QQ.hs:119]: 👍

File: .hlint.yaml [hunk 10]
```diff
@@ -0,0 +1,2 @@
+- arguments:
+    - --ignore-glob=README.lhs
```
Comment (sol) [inline .hlint.yaml:2]: Trailing newline please.

**Review by sol — CHANGES_REQUESTED** [review]

**Comment by sol** [comment]
@philderbeast note that when you don't need interpolation then `yamlQQ` from `Data.Yaml.TH` is a more lightweight option.

File: .hlint.yaml (hunk 10 above)
Comment (philderbeast) [reply .hlint.yaml]: Added the newline.

File: src/Data/JSON/QQ.hs (hunk 4 above)
Comment (philderbeast) [reply src/Data/JSON/QQ.hs:129]: > 4.7.0.0 Apr 2014
> ...
> Data.Functor now exports ($>) and void.
> SOURCE: [base changelog](https://hackage.haskell.org/package/base-4.18.0.0/changelog)

Comment (philderbeast) [reply src/Data/JSON/QQ.hs:129]: I bumped base to match.

File: test/Data/JSON/QQSpec.hs [hunk 11]
```diff
@@ -12,39 +12,32 @@ spec :: Spec
...
 
     it "fails on excess input" $ do
-      let Left err = parsedJson "{foo: 23} some excess input"
-      show err `shouldBe` "\"txt\" (line 1, column 11):\nunexpected 's'\nexpecting space or end of input"
+      parsedJson "{foo: 23} some excess input" `shouldSatisfy` either
+        ((== "\"txt\" (line 1, column 11):\nunexpected 's'\nexpecting space or end of input") . show)
+        (const False)
```
[the remark this answers is not in the document]
Comment (philderbeast) [reply test/Data/JSON/QQSpec.hs:43]: I reverted this but didn't add the `OPTIONS_GHC` as I see you suppressed this warning in the cabal file. I'd felt uncomfortable suppressing the warning myself (as was my first instinct).

File: test/Data/JSON/QQSpec.hs (hunk 1 above)
Comment (philderbeast) [reply test/Data/JSON/QQSpec.hs:38]: Good to know, thanks. I added this constraint to the test suite.

**Review by sol — APPROVED** [review]
This looks great 👍

**Pull request merged**